File: app/src/main/java/com/d4rk/androidtutorials/java/ui/android/alerts/alertdialog/AlertDialogActivity.java
package com.d4rk.androidtutorials.java.ui.android.alerts.alertdialog;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;

import com.d4rk.androidtutorials.java.R;
import com.d4rk.androidtutorials.java.databinding.ActivityAlertDialogBinding;
import com.d4rk.androidtutorials.java.utils.EdgeToEdgeDelegate;
import com.google.android.material.dialog.MaterialAlertDialogBuilder;

public class AlertDialogActivity extends AppCompatActivity {
    private final Handler handler = new Handler();
    private ActivityAlertDialogBinding binding;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityAlertDialogBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        EdgeToEdgeDelegate edgeToEdgeDelegate = new EdgeToEdgeDelegate(this);
        edgeToEdgeDelegate.applyEdgeToEdge(binding.container);

        MaterialAlertDialogBuilder alertDialog = createAlertDialog();
        binding.button.setOnClickListener(v -> alertDialog.show());
        binding.floatingButtonShowSyntax.setOnClickListener(v -> startActivity(new Intent(this, AlertDialogCodeActivity.class)));
        handler.postDelayed(() -> binding.floatingButtonShowSyntax.shrink(), 5000);
    }

    private MaterialAlertDialogBuilder createAlertDialog() {
        return new MaterialAlertDialogBuilder(this)
                .setTitle(R.string.your_title)
                .setMessage(R.string.summary_alert_dialog_message)
                .setIcon(R.drawable.ic_shop)
                .setPositiveButton(android.R.string.ok, null)
                .setNegativeButton(android.R.string.cancel, null);
    }
}

File: app/src/main/kotlin/com/d4rk/androidtutorials/java/ui/android/alerts/alertdialog/AlertDialogActivity.kt
package com.d4rk.androidtutorials.java.ui.android.alerts.alertdialog
import android.content.Intent
import android.os.Bundle
import android.os.Handler
import androidx.appcompat.app.AppCompatActivity
import com.d4rk.androidtutorials.java.R
import com.d4rk.androidtutorials.java.databinding.ActivityAlertDialogBinding
import com.google.android.material.dialog.MaterialAlertDialogBuilder
class AlertDialogActivity : AppCompatActivity() {
    private lateinit var binding: ActivityAlertDialogBinding
    @Suppress("DEPRECATION")
    private val handler = Handler()
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityAlertDialogBinding.inflate(layoutInflater)
        setContentView(binding.root)
        val alertDialog = createAlertDialog()
        binding.button.setOnClickListener {
            alertDialog.show()
        }
        binding.floatingButtonShowSyntax.setOnClickListener {
            startActivity(Intent(this, AlertDialogCodeActivity::class.java))
        }
        handler.postDelayed({
            binding.floatingButtonShowSyntax.shrink()
        }, 5000)
    }
    private fun createAlertDialog(): MaterialAlertDialogBuilder {
        return MaterialAlertDialogBuilder(this).apply {
            setTitle(R.string.title_alert_dialog)
            setMessage(R.string.alert_dialog_message)
            setIcon(R.drawable.ic_shop)
            setPositiveButton(android.R.string.ok, null)
            setNegativeButton(android.R.string.cancel, null)
        }
    }
}